gnucash/gnome/gnc-split-reg.c

```diff
@@ -2152,11 +2152,14 @@ gnc_split_reg_set_sort_reversed(GNCSplitReg *gsr, gboolean rev, gboolean refresh
      *       In other words, qof_query_set_sort_increasing should
      *       always use the inverse of rev.
      */
+    SplitRegister *reg = gnc_ledger_display_get_split_register (gsr->ledger);
     Query *query = gnc_ledger_display_get_query( gsr->ledger );
+
+    gnc_split_register_set_reverse_sort (reg, rev);
+
     qof_query_set_sort_increasing (query, !rev, !rev, !rev);
     gsr->sort_rev = rev;
-    Account *acct = gnc_ledger_display_leader (gsr->ledger);
-    xaccAccountSetSortReversed(acct, rev);
```

Comment:
Should the calls to `xaccAccount[GS]etSortReversed` in `gsr_move_sort_and_filter_to_state_file` be similarly replaced? What about `xaccAccount[GS]etSortOrder` in that function?

That's the only use for either of those functions, and the corresponding QOF properties are similarly unused so ISTM they should be removed in favor of split register state.

That's out of scope for this PR and should probably go in master.